prow/config/config.go

```diff
@@ -1023,11 +1041,30 @@ func (c *Config) validateComponentConfig() error {
 		}
 	}
 
+	// TODO(@clarketm): Remove in May 2020
 	if c.SlackReporter != nil {
-		if err := c.SlackReporter.DefaultAndValidate(); err != nil {
-			return fmt.Errorf("failed to validate slackreporter config: %v", err)
+		logrus.Warning("slack_reporter will be deprecated on May 2020, and it will be replaced with slack_reporter_configs['*'].")
+
+		if c.SlackReporterConfigs == nil {
+			c.SlackReporterConfigs = map[string]*SlackReporter{}
+		}
+
+		if c.SlackReporterConfigs["*"] != nil {
```

Comment:
Nit: I would make the whole settings mutually exclusive and not just `SlackReporterConfigs["*"]` and `SlackReporterConfig`

Reply:
Now if both `c.SlackReporter` and `c.SlackReporterConfigs` exist, an `error` is thrown.